### ISSUE TITLE
Training Cascade Mask RCNN with Mutliple-GPU.

### ISSUE THREAD

**Issue by IAMShashankk** [open-mmlab/mmdetection#5795]
**Describe the bug**

When I am launching training job for Cascade Mask RCNN on 8 GPU (or >1 GPU) I am facing error before training starts.
My dataset works perfectly for a single GPU.
**Reproduction**

1. What command or script did you run?
```
bash tools/dist_train.sh /netscratch/mishra/Test_Project/new_configs/resnet101_cascade_mask_rcnn_florplan_config_fine_tuned.py 4
```

2. Did you make any modifications to the code or config? Did you understand what you have modified?
Modification is only done in the config file as per my dataset.

3. What dataset did you use?
Custom dataset.

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
```
Collecting environment information...
PyTorch version: 1.9.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.4 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.14.0
Libc version: glibc-2.9

Python version: 3.6.10 |Anaconda, Inc.| (default, Mar 23 2020, 23:13:11)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-77-generic-x86_64-with-debian-buster-sid
Is CUDA available: True
CUDA runtime version: 11.0.167
GPU models and configuration: 
GPU 0: Tesla V100-SXM2-16GB
GPU 1: Tesla V100-SXM2-16GB
GPU 2: Tesla V100-SXM2-16GB
GPU 3: Tesla V100-SXM2-16GB

Nvidia driver version: 460.73.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip] crumpets-torch==3.0.0a5+cuda110
[pip] msgpack-numpy==0.4.3.2
[pip] numpy==1.18.1
[pip] pytorch-ignite==0.4.1
[pip] pytorch-transformers==1.1.0
[pip] torch==1.9.0
[pip] torch-geometric==1.6.0
[pip] torchaudio==0.9.0
[pip] torchtext==0.6.0
[pip] torchvision==0.10.0
[conda] crumpets-torch            3.0.0a5+cuda110          pypi_0    pypi
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.1                      144  
[conda] mkl-include               2019.1                      144  
[conda] msgpack-numpy             0.4.3.2                  py36_0  
[conda] nomkl                     3.0                           0  
[conda] numpy                     1.18.1           py36h94c655d_0  
[conda] numpy-base                1.18.1           py36h2f8d375_1  
[conda] pytorch-ignite            0.4.1                    pypi_0    pypi
[conda] pytorch-transformers      1.1.0                    pypi_0    pypi
[conda] torch                     1.9.0                    pypi_0    pypi
[conda] torch-geometric           1.6.0                    pypi_0    pypi
[conda] torchaudio                0.9.0                    pypi_0    pypi
[conda] torchtext                 0.6.0                    pypi_0    pypi
[conda] torchvision               0.10.0                   pypi_0    pypi
```
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.

```
[W ProcessGroupNCCL.cpp:1569] Rank 0 using best-guess GPU 0 to perform barrier as devices used by this process are currently unknown. This can potentially cause a hang if this rank to GPU mapping is incorrect.Specify device_ids in barrier() to force use of a particular device.
/opt/conda/lib/python3.6/multiprocessing/semaphore_tracker.py:143: UserWarning: semaphore_tracker: There appear to be 1 leaked semaphores to clean up at shutdown
  len(cache))
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: -9) local_rank: 1 (pid: 3114573) of binary: /opt/conda/bin/python
ERROR:torch.distributed.elastic.agent.server.local_elastic_agent:[default] Worker group failed
INFO:torch.distributed.elastic.agent.server.api:[default] Worker group FAILED. 3/3 attempts left; will restart worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Stopping worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous'ing worker group
INFO:torch.distributed.elastic.agent.server.api:[default] Rendezvous complete for workers. Result:
  restart_count=1
  master_addr=127.0.0.1
  master_port=29511
  group_rank=0
  group_world_size=1
  local_ranks=[0, 1, 2, 3]
  role_ranks=[0, 1, 2, 3]
  global_ranks=[0, 1, 2, 3]
  role_world_sizes=[4, 4, 4, 4]
  global_world_sizes=[4, 4, 4, 4]

INFO:torch.distributed.elastic.agent.server.api:[default] Starting worker group
INFO:torch.distributed.elastic.multiprocessing:Setting worker0 reply file to: /tmp/torchelastic_x6pry4n_/none_v4lc99ii/attempt_1/0/error.json
INFO:torch.distributed.elastic.multiprocessing:Setting worker1 reply file to: /tmp/torchelastic_x6pry4n_/none_v4lc99ii/attempt_1/1/error.json
INFO:torch.distributed.elastic.multiprocessing:Setting worker2 reply file to: /tmp/torchelastic_x6pry4n_/none_v4lc99ii/attempt_1/2/error.json
INFO:torch.distributed.elastic.multiprocessing:Setting worker3 reply file to: /tmp/torchelastic_x6pry4n_/none_v4lc99ii/attempt_1/3/error.json
```

The error only occurs when I try to run on multiple GPUs.
I tried starting training like below but no luck.
```
CUDA_VISIBLE_DEVICES = 1,2,3,4 tools/dist_train.sh /netscratch/mishra/Test_Project/new_configs/resnet101_cascade_mask_rcnn_florplan_config_fine_tuned.py 4

Coul you please let me know how to make it work with more than one GPU.

**Comment by RangiLyu** [comment]
The command you run is correct. Could you provide more detailed error messages?

**Comment by alaa-shubbak** [comment]
I am running the similar code of cascade mask rcnn , i wonder if i can save an evaluation matrix for mAP for mask and bbox ? 
I am trying to get such value and plot them using the function: 

`python tools/analysis_tools/analyze_logs.py plot_curve `

for each epoch it save the losses values only , not the mAP as in this image: 
![casacde mask rcnn](https://user-images.githubusercontent.com/10245810/128924336-14a3f46e-9044-4ea8-a9b1-4b6f3208d6ab.JPG)

how can i save the mAP values to choose the best epoch for do the inferences part  (test )? looking for any suggestion please

**Comment by alaa-shubbak** [comment]
![testing results](https://user-images.githubusercontent.com/10245810/128933835-70f7acc2-a391-4637-85eb-f2ee4d2a1676.JPG)

here the message of the results of testing the data

**Comment by RangiLyu** [comment]
> I am running the similar code of cascade mask rcnn , i wonder if i can save an evaluation matrix for mAP for mask and bbox ? I am trying to get such value and plot them using the function:
> 
> `python tools/analysis_tools/analyze_logs.py plot_curve `
> 
> for each epoch it save the losses values only , not the mAP as in this image: ![casacde mask rcnn](https://user-images.githubusercontent.com/10245810/128924336-14a3f46e-9044-4ea8-a9b1-4b6f3208d6ab.JPG)
> 
> how can i save the mAP values to choose the best epoch for do the inferences part (test )? looking for any suggestion please

Your loss is NAN. So there are no eval results. Please follow the "Loss goes NAN" in this doc https://github.com/open-mmlab/mmdetection/blob/master/docs/faq.md
